aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp

```diff
@@ -229,19 +259,17 @@ namespace Aws
 
 
             OpenSSLCipher::OpenSSLCipher(const CryptoBuffer& key, size_t blockSizeBytes, bool ctrMode) :
-                    SymmetricCipher(key, blockSizeBytes, ctrMode), m_encDecInitialized(false), m_encryptionMode(false),
-                    m_decryptionMode(false)
+                    SymmetricCipher(key, blockSizeBytes, ctrMode), m_ctx(nullptr),
+                    m_encDecInitialized(false), m_encryptionMode(false), m_decryptionMode(false)
             {
                 Init();
             }
 
             OpenSSLCipher::OpenSSLCipher(OpenSSLCipher&& toMove) : SymmetricCipher(std::move(toMove)),
-                                                                   m_encDecInitialized(false)
+                                                                   m_ctx(nullptr), m_encDecInitialized(false)
             {
-                m_ctx = toMove.m_ctx;
-                toMove.m_ctx.cipher = nullptr;
-                toMove.m_ctx.cipher_data = nullptr;
-                toMove.m_ctx.engine = nullptr;
+                Init();
+                EVP_CIPHER_CTX_copy (m_ctx, toMove.m_ctx);
 
```

Comment:
In original impl, there are several fields set up to nullptr, and I checked the code of EVP_CIPHER_CTX_copy, it doesn't cleanup in arg, I guess we should call EVP_CIPHER_CTX_cleanup(toMove.m_ctx) here.